ffi/Cargo.toml

```
@@ -52,9 +44,7 @@ default = ["default-engine"]
 cloud = ["delta_kernel/cloud"]
```

Also need: (won't quite apply cleanly here because github is stupid about letting me comment on "unchanged" lines)

```suggestion
arrow = ["arrow_53"]
arrow_53 = ["dep:arrow-data_53", "dep:arrow-schema_53"]
arrow_54 = ["dep:arrow-data_54", "dep:arrow-schema_54"]

default = ["arrow", "default-engine"]
cloud = ["delta_kernel/cloud"]
```